File: CRUD_UI/ProductUI/src/app/shared/entities/product.ts
import {ObjectStateType} from "./enums";
import {productPrice} from "../../shared/entities/productPrice";

export class Product {
  productId: number;
  name: string;
  description: string;
  active: boolean;
  objectState: ObjectStateType;
  productPrices: productPrice[] = [];
  constructor() {
    this.active = false;
  }
}

File: Hybrid-Microservices-UI/CrudApi_Microservices/Product.Entities/Product.cs
using CRUD.Transaction.CRUDApi.Core.Entities;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Runtime.Serialization;


namespace CRUD.Transaction.Product.Entities
{
    [DataContract(IsReference = true)]
    [KnownType(typeof(ModelEntityBase))]
    [KnownType(typeof(List<ProductPrice>))]
    public partial class Product : ModelEntityBase
    {
        public Product()
        {
            this.ProductPrices = new List<ProductPrice>();
    
        }

        #region base properties
        [DataMember]
        public long ProductId { get; set; }
        [DataMember]
        public string Name { get; set; }
        [DataMember]
        public string Description { get; set; }
        [DataMember]
        public bool Active { get; set; }
        #endregion

        #region custom or navigation properties
         
        [DataMember]
        [ChildNavigationEntity]
        public List<ProductPrice> ProductPrices { get; set; }


        #endregion

    }
}
